### ISSUE TITLE
pyinstaller should not crash when hitting an encoding error in source code

### ISSUE THREAD

**Issue by y2kbugger** [pyinstaller/pyinstaller#2212]
python=3.5
PyInstaller=3.2

```
file_handle = StringIO(loader.get_source(module_name))
File "<frozen importlib._bootstrap_external>", line 719, in get_source
File "<frozen importlib._bootstrap_external>", line 505, in decode_source
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 452: invalid start byte
```

Error occurs when building and is uncaught.

It was caused by the same improperly encoded "ENDASH" as the error I reported here that also hung up  pyparsing.

http://stackoverflow.com/questions/38884540/why-am-i-getting-this-error-from-having-nonvalid-unicode-in-source-but-only-whe

**Comment by htgoebel** [comment]
And what is the error you want to report? Of course if you input garbage, the program crashes.

**Comment by y2kbugger** [comment]
pyinstaller _should not_ crash when hitting an encoding error.
It _could_ fail and give an explanation of the error.
It _could_ ignore the encoding issue as it is in a comment—If the python interpreter itself doesn't choke, neither should pyinstaller.

For what it's worth, pyparsing was fixed in their next release.

As a first time user of pyinstaller, I had know idea where the error came from (my code?, a library?, a fault in pyinstaller itself?) The only reason I figured it out is that I remembered it from before. People paste various text from the web into comments frequently.
My intent was not to get a fix from you for my sake, but help the software give a better user experience for others in the future.

**Comment by htgoebel** [comment]
Thanks for the explanation, I updated the title accordingly.

Re 
> If the python interpreter itself doesn't choke, neither should pyinstaller.

We are just using the python code there. So this is out of our capabilities.

This is how it could be implemented: Wrap `file_handle = StringIO(loader.get_source(module_name))` and catch encoding errors there. On Error, raise system exit and an error message stating the module-name and if possible the file-name.

**Comment by arthurspa** [comment]
Isn't this issue already fixed by the `except UnicodeDecodeError` in f8e8ae93?

**Comment by htgoebel** [comment]
Yes, I assume this is already fixed.
